### PR TITLE
Watch for registration status in the router

### PR DESCRIPTION
**Pull request by Flaburgan**

### CODE REVIEW

**Comment by Flaburgan** [comment]
@nanu-c so, did you try all the use cases here, registration, password, pin etc? If everything works then I'll squash and merge this.

File: axolotl-web/src/router/router.js [hunk 1]
```diff
@@ -96,42 +102,47 @@ export const router = new Router({
...
   }
+});
 
-  // We have a status (regisered or not). Let's check access right
-  const publicPages = ['/register', '/verify', '/password'];
-  const authRequired = !publicPages.includes(to.path);
+function proceed(to, next) {
+  const registrationPages = ['/register', '/verify', '/password', '/pin'];
+  const registrationStatus = store.state.registrationStatus;
+  store.isRegistrationStatusUpToDate = false;
```
Comment (Flaburgan) [inline axolotl-web/src/router/router.js:126]: I don't know where to put this. Leaving it there means a call to the server will be done each time we request a route.

Comment (nanu-c) [reply axolotl-web/src/router/router.js:126]: Watching the `SET_REGISTRATION_STATUS` in the vuex store, it's only sent twice from the backend. No matter how many chats or settings pages I open.

Comment (Flaburgan) [reply axolotl-web/src/router/router.js:126]: Yes there is something not working as I expect there, that's also probably the reason why I got stuck once. Do I need to "commit" or do something like this to actually change the value in the store?

Comment (nanu-c) [reply axolotl-web/src/router/router.js:126]: yup, vuex uses the flux principle and therefore the database/vuex-storage works like git. Every change has to be committed.

Comment (nanu-c) [reply axolotl-web/src/router/router.js:126]: You need a mutation in store.js like:
```
    SET_REGESTRATION_STATUS_UP_TO_DATE(state, status) {
      store.isRegistrationStatusUpToDate = status;
    },
```
and then you can do something like:
```
store.commit("SET_REGESTRATION_STATUS_UP_TO_DATE", false)
```

File: axolotl-web/src/store/store.js [hunk 2]
```diff
@@ -325,6 +326,9 @@ export default new Vuex.Store({
         window.location.replace("/")
       }
     },
+    SET_REGESTRATION_STATUS_UP_TO_DATE(state, status) {
```
Comment (Flaburgan) [inline axolotl-web/src/store/store.js:329]: Typo, REGISTRATION

**Review by nanu-c — APPROVED** [review]
Looks good to me, I tested registration, verification and encrypted database and all of them are working

**Pull request merged**